src/nsidc/metgen/metgen.py

```diff
@@ -297,21 +299,20 @@ def process(configuration: config.Config) -> None:
     summarize_results(results)
 
 
-def data_reader(data_files: set[str]) -> Callable[[str], dict]:
+def data_reader(
+    collection: str, data_files: set[str]
+) -> Callable[[str, config.Config], dict]:
     """
     Determine which file reader to use for the given data files. This currently
-    is limited to handling one data file type, and one reader. In a future
-    issue, we may handle granules with multiple data file types per granule.
+    is limited to handling one data file type (and one reader) per collection.
+    In a future issue, we may handle granules with multiple data file types per granule.
     In that future work this needs to be refactored to handle this case.
     """
-    readers = {
-        ".nc": netcdf_reader.extract_metadata,
-        ".csv": csv_reader.extract_metadata,
-    }
-
-    _, extension = os.path.splitext(first(data_files))
+    # Lookup based on an arbitrary data file in the set
+    a_data_file = next(iter(data_files))
```

Comment:
Why not use `first` here (as in `derived_granule_name`)?

Reply:
I can only guess that it is because of the quantum wave function. But in any case, `first` is so much better.